server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/GlobalOrdinalValuesSource.java

```diff
@@ -214,4 +291,170 @@ private void initLookup(SortedSetDocValues dvs) throws IOException {
             }
         }
     }
+
+    public void updateHighestCompetitiveValue(int slot) throws IOException {
+        highestCompetitiveValueGlobalOrd = values.get(slot);
+        logger.trace("Highest observed set to [{}]", highestCompetitiveValueGlobalOrd);
+        final CompetitiveIterator competitiveIterator = currentCompetitiveIterator;
+        if (competitiveIterator != null) {
+            competitiveIterator.updateBounds();
+        }
+    }
+
+    void collectDebugInfo(BiConsumer<String, Object> add) {
+        add.accept("num_global_ordinal_dynamic_pruning_initialized", currentCompetitiveIterator == null ? 0 : dynamicPruningInitialized);
```

Comment:
nit: maybe also add the total number of segments seen, so that it's possible to know whether it was used on all segments or just a small subset?

Reply:
Effectively `num_global_ordinal_dynamic_pruning_initialized` is the number of segments. We create one `LeafBucketCollector` per segment, thus potentially up to one competitive iterator per segment. And we count the number of times it actually gets initialized. Perhaps I should rename this to something like `num_segments_global_ordinal_dynamic_pruning_used`?

Reply:
Sorry I meant to track the total number of segments seen by this aggregator, including those where init() is never called with something like `segments_collected` as a key (consistent with FilterByFilterAggregator). As far as naming is concerned, it looks like a few other aggregators adopted the naming convention of prefixing segment counts with `segments_` (e.g. FilterByFilterAggregator), so `segments_dynamic_pruning_used`? (I made it a bit shorted by removing references to global ords since the terms source only works with global ords anyway?)

Reply:
I see. OK, will do that.

As far as naming is concerned, I thought having `global_ordinals` in there would distinguish this debug info from other debug info that could potentially come from a nested aggregation. But I'll do the changes anyway and if we have a name clash in the future we can resolve it then.

Reply:
Another idea would be to prefix with `sources.{terms_source_name}.segments_dynamic_pruning_used` and `sources.{terms_source_name}.segments_collected`. What do you think?

Reply:
> distinguish this debug info from other debug info that could potentially come from a nested aggregation

Inner aggregations should put their debug info in their own namespace, so we don't need to worry about naming collisions.

Adding the name of the source to the debug info makes sense to me. Tagging @martijnvg in case he has thoughts on this.

Reply:
Agreeing with this statement here.